R/setkey.R

```diff
@@ -353,6 +354,9 @@ CJ <- function(..., sorted = TRUE, unique = FALSE)
                 l[[i]] = rep.int(rep.int(y, times = rep.int(x[i], n[i])), times = nrow/(x[i]*n[i]))
             if (any(class(l[[i]]) != j[[i]]))
                 setattr(l[[i]], 'class', j[[i]]) # reset "Date" class - rep.int coerces to integer
+            if (any(!sapply(tzones, is.null))) {
```

Comment:
The difference to the line above is, that it tests for each `i`. 
So, if we want to retain the tzone if-statement, it should read:
`if(!is.null(tzones[[i]])){...}`. That would make more sense in my opinion.